src/Equinox.Cosmos/Cosmos.fs

```diff
@@ -447,20 +440,62 @@ function sync(req, expIndex, expEtag) {
     }
 }"""
 
+module CancellationToken =
+    let useOrCreate = function Some ct -> async.Return ct | _ -> Async.CancellationToken
+
+type EquinoxCosmosClient (cosmosClient: CosmosClient, containerClient: CosmosContainer, databaseId: string, containerId: string) =
```

Comment:
Hmmm, not sure if I agree on this. It's a type that acts as Equinox's interface to Cosmos operations. Wouldn't Client be the right terminology here? I don't think it's Context, because that term implies that there is some state being held. This type isn't holding any state.

Reply:
I'm thinking of how the name overlaps with the Azure.Cosmos.CosmosClient - one might think that an something targeting Azure.Cosmos (Equinox) that has a thing called EquinoxX would be a fairly simple decorator for thing X in Azure.Cosmos

So my concern is that the name is lying - it's doing much more than the `CosmosClient` bit suggests (although one can take the view that it's `EquinoxCosmos`'s Client wrapper)

I think both the fact we are using Equinox in a name, and are wrapping more than CosmosClient can be addressed by a light name change.

Reply:
But wouldn't the `Equinox` in front of the name prevent consumers from thinking that it is part of `Azure.Cosmos.CosmosClient`? This is similar to how Cosmos DB has a `CosmosJsonSerializer` type, which is an interface into an underlying `JsonSerializer`. 

What other names do you have in mind?

Reply:
I don't have great suggestions for other names... yet
The contrast with the same stuff in the Cosmos SDK is that there are exactly _zero_ usages of the string `Equinox` in any name (as opposed to namespace) so far _combined with the fact that it is not a direct wrapper for_ only `CosmosClient` - the previous name (with Container in it), is less confusing as it implies that we're talking about a specific Container in a specific Database, whereas `CosmosClient` makes me think we are talking at the Account level

as mentioned in the initial comment, some opening suggestions are:
- given this does more than just wrap a Client, it seems that something without Client in the name (Context, Store ?)
- maybe juggle the names of other things like Gateway and Context and play with actual usage in e.g. Cosmos.fsx ?

I think my best suggestion is `CosmosStore` or `CosmosContext` - I'd be OK with either of those being prefixed with `Equinox`, but don't necessarily see the value given that it doesnt need disambiguation to the same degree?

Reply:
As I mentioned above, I think that `Context` isn't appropriate because that word implies a stateful type that is alterable, which this is not.

As for `Store`, I also think that it's unintuitive because it implies some storage of data / information, which this is not either.

If you're very against the term `Client`, would you maybe be ok with `Operations`?

Also, I would suggest we keep the `Equinox` at the front of the type since this is a type that will be used directly by consumers, and `CosmosClient` / `CosmosOperations` / `CosmosStore` / etc makes it sound like it is part of the Cosmos SDK itself, no?

Reply:
yes, Operations would be fine too (Wrapper springs to mind) - my primary desire would be to avoid clashing with things that it's much broader than; especially when they are Domain UL terms, if you will (that domain being Azure.Cosmos)

having Cosmos in the name is OK (but would make more sense if we put Cosmos back onto all the names as Cosmos V4 has done (but then that would beg the question of whether similar prefixing should happen in all the other store bindings too))

So Equinox.Cosmos.Internal.CosmosOperations, StoreContext, ContainerContext are the sort of names entering my head atm

but my primary concern was the use of `CosmosClient` when it's more than that

Reply:
Renamed `EquinoxCosmosClient*` to `EquinoxCosmosOperations*`.